### PR TITLE
mining: add visual icon/indicator that mining is enabled, visible on all the screens

### LINKED ISSUE #661: [FEATURE REQUEST] "Show some sort of visual icon/indicator that mining is enabled visible on all the screens"
From [Reddit](https://www.reddit.com/r/Monero/comments/62e1mc/mandatory_upgrade_monero_01031_wolfram/dftatcf/?context=3):

>gui suggestion - if mining is enabled in the gui. Show some sort of visual icon/indicator that mining is enabled visible on all the screens. Perhaps down the bottom where Network Status/Lightning Bolt is. Nice work.

### PR DESCRIPTION
**Pull request by mmbyday**
Resolves #661 
- icon is clickable and goes to mining page

mining222

### CODE REVIEW

**Comment by MaxXor** [comment]
What about a pickaxe icon instead of gold bars?

**Comment by mmbyday** [comment]
Sure @MaxXor . Open to suggestions. Here's some ideas to start with https://emojipedia.org/pick/

**Comment by MaxXor** [comment]
I don't mind which one to choose. Just be sure that the icon is under a free license.

**Comment by xmrdsc** [comment]
I think changing the lightning bolt icon (left side) to an orange pickaxe would look nice. Less intrusive too.

**Comment by mmbyday** [comment]
Thinking about using one from 
https://www.emojione.com/licenses

Any advice on how to handle attributions for their Free License?
Do we fall under Attribution Required > Commercial use only?? 

@xmrdsc agreed on the lightning bolt swap. bloody brilliant. or maybe do you know how we can source a orange pickaxe?

**Comment by erciccione** [comment]
@mmbyday From the license:

> Personal projects—those that are not commercial in nature—are permitted to use the JoyPixels/EmojiOne Artwork without attribution
(we’d like you to mention us in some way, but it isn’t required for personal projects)

We are not a commercial entity since we are not selling a product.

I went through the license and i found no conflicts. We can use it with no issues.

**Comment by mmbyday** [comment]
updated.

pick

File: main.qml [hunk 1]
```diff
@@ -72,6 +72,7 @@ ApplicationWindow {
     property bool qrScannerEnabled: (typeof builtWithScanner != "undefined") && builtWithScanner
     property int blocksToSync: 1
     property var isMobile: (appWindow.width > 700 && !isAndroid) ? false : true
+    property bool isMining : false
```
Comment (xiphon) [inline main.qml]: Just a minor styling comment
```suggestion
    property bool isMining: false
```

File: components/NetworkStatusItem.qml [hunk 2]
```diff
@@ -68,17 +68,46 @@ Rectangle {
...
+                anchors.top: parent.top
+                anchors.topMargin: 4 * scaleRatio
+                anchors.right: parent.right
+                source: "../images/miningxmr.png"
+                width: 32 * scaleRatio
+                height: 32 * scaleRatio
+                MouseArea {
+                    anchors.fill: parent
+                    cursorShape: Qt.PointingHandCursor
+                    hoverEnabled: true
```
Comment (xiphon) [inline components/NetworkStatusItem.qml]:
```suggestion
```

File: components/NetworkStatusItem.qml [hunk 3]
```diff
@@ -68,17 +68,46 @@ Rectangle {
...
                     if(item.connected == Wallet.ConnectionStatus_Connected){
                         return "../images/lightning.png"
                     } else {
                         return "../images/lightning-white.png"
                     }
                 }
+                MouseArea {
+                    anchors.fill: parent
+                    cursorShape: Qt.PointingHandCursor
+                    hoverEnabled: true
```
Comment (xiphon) [inline components/NetworkStatusItem.qml]:
```suggestion
```

File: components/NetworkStatusItem.qml [hunk 4]
```diff
@@ -68,17 +68,46 @@ Rectangle {
...
             Image {
+                visible: !appWindow.isMining
                 anchors.top: parent.top
-                anchors.topMargin: 6
+                anchors.topMargin: 6 * scaleRatio
                 anchors.right: parent.right
-                anchors.rightMargin: 11
+                anchors.rightMargin: 11 * scaleRatio
                 source: {
                     if(item.connected == Wallet.ConnectionStatus_Connected){
```
Comment (xiphon) [inline components/NetworkStatusItem.qml]: Better to have single Image instance and add update `source` and `onClicked` to handle the mining status changes

Comment (mmbyday) [reply components/NetworkStatusItem.qml]: Noted. Thanks.

**Comment by dEBRUYNE-1** [comment]
Any opinions on whether the pickaxe should be on the left or right side (UI wise)?

**Comment by xiphon** [comment]
> Any opinions on whether the pickaxe should be on the left or right side (UI wise)?

https://github.com/monero-project/monero-gui/pull/1830#issuecomment-448403428 looks good to me.

**Pull request merged**